src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -803,7 +803,7 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
     }
 
     void process_dims(Allocator &al, Vec<ASR::dimension_t> &dims,
-        AST::dimension_t *m_dim, size_t n_dim) {
+        AST::dimension_t *m_dim, size_t n_dim, int &is_compile_time) {
```

Comment:
```suggestion
        AST::dimension_t *m_dim, size_t n_dim, bool &is_compile_time) {
```
Also it is not initialized in all cases. You have to set `is_compile_time = false` at the beginning of the function.